libs/remote_storage/src/lib.rs

```diff
@@ -205,6 +205,28 @@ pub trait RemoteStorage: Send + Sync + 'static {
     async fn delete(&self, path: &RemotePath) -> anyhow::Result<()>;
 
     async fn delete_objects<'a>(&self, paths: &'a [RemotePath]) -> anyhow::Result<()>;
+
+    /// Copy a remote object inside a bucket from one path to another.
+    async fn copy(&self, from: &RemotePath, to: &RemotePath) -> anyhow::Result<()> {
+        // this is an inefficient generic implementation
+
+        let download = self.download(from).await?;
+        let mut reader = tokio_util::io::StreamReader::new(download.download_stream);
+
+        let camino_file = camino_tempfile::tempfile()?;
```

Comment:
No, objection. This will download into `/tmp`, which is a slow EBS volume on Safekeepers. You don't want to do that.

Reply:
Note that this code is only for the case when underlying `RemoteStorage` hasn't implemented `copy` manually. This applies only to Azure and tests currently, because I provided a fast implementation for the S3.

Any ideas how to make it better? I can't just glue download+upload stream together because I need to know the size in advance...

Reply:
I'm fine with a tempfile. But, it should sit on `/storage` , not in `/tmp`.

Or, whatever the equivalent of `/storage` is when we run tests.

Reply:
I think at a minimum, don't have a blanket default impl, but, force each impl to think about this.

Reply:
I think it'd be fine if the azure impl returns an error.

LocalFs can use some `.tmp` directory in its root path

Reply:
> I can't just glue download+upload stream together because I need to know the size in advance...

We could maybe add a length field to the `Download` struct? HTTP requires the `content-length` header to be present in most cases, and I think in the cases we use S3/Azure for, it's always sent.

Reply:
> I think it'd be fine if the azure impl returns an error.
>
> LocalFs can use some .tmp directory in its root path

:+1: to that as well. IMO we could always just error.

Reply:
Removed generic implementation with tmp, LocalFs imlpementation is a usual fs::copy now.